sympy/stats/crv_types.py

```diff
@@ -2870,6 +2878,19 @@ def _quantile(self, p):
         mean, std = self.mean, self.std
         return mean + std*sqrt(2)*erfinv(2*p - 1)
 
+    def sample(self, size=(), library='scipy'):
+        if self.std == 0:
+            # Dirac Delta is not implemented in scipy so we must do sampling another way
```

Comment:
It might make sense to at least open a scipy and/or numpy issue about allowing zero variance, so you can reference it here.

Reply:
Looks like numpy and scipy can both handle 0 variance:
```python
import numpy as np
import scipy.stats as st

mu, sigma = 1, 0  # mean and standard deviation
s = np.random.normal(mu, sigma, 10)
print(s)  # [1. 1. 1. 1. 1. 1. 1. 1. 1. 1.]

r = st.norm.rvs(size=10, loc=mu, scale=sigma)
print(r)  # [1. 1. 1. 1. 1. 1. 1. 1. 1. 1.]
```

What's going on is that the `sample` method seems to create a handmade distribution for numpy and scipy. But many are already implemented by default. 

This means that both numpy and scipy need to numerically integrate a function that's literally 0 everywhere except at a point. I also wouldn't expect libraries like numpy and scipy to implement a super theoretical function like `DiracDelta`.

I'll see if I can change the `sample` method to match some predefined distributions between the various libraries.

Reply:
Are you sure? I thought the sample method just used a lookup table to find the distribution functions you call there..

Reply:
I think sampling was recently changed (#19342). It works as you say for numpy. In fact, I just tried commenting out my `sample` override of the `NormalDistribution` and ran the following:
```python
Y = Normal("Y", 0, 0)
for x in sample_iter(Y, numsamples=5, library='numpy'):
    print(x)
```
Works just fine.

Scipy doesn't seem to have a map like that yet. I'll just add one since it's almost a copy of the numpy version.